### ISSUE TITLE
Host reaches DOWN/HARD with current_problem_id=0 after parent SOFT DOWN/UP causes UNREACHABLE->DOWN transition

### ISSUE THREAD

**Issue by rjwesten** [NagiosEnterprises/nagioscore#1061]
Nagios Core version: 4.5.12
OS: Red Hat Enterprise Linux release 8.10 (Ootpa)
Install method: source compiled

Problem:
A host reached DOWN/HARD but status.dat and retention.dat show current_problem_id=0.

Expected:
For a host in a hard DOWN problem state, current_problem_id should be non-zero.

Actual:
Host is DOWN/HARD, notifications were sent, but current_problem_id is 0.

Relevant child host log:

[1780696009] HOST ALERT: MTPC;UNREACHABLE;SOFT;1;CRITICAL - Socket timeout
[1780696072] HOST ALERT: MTPC;DOWN;SOFT;2;CRITICAL - Socket timeout
[1780696135] HOST ALERT: MTPC;DOWN;SOFT;3;connect to address 172.28.54.2 and port 50005: No route to host
[1780696198] HOST NOTIFICATION: operations;MTPC;DOWN;host-notify-operations;connect to address 172.28.54.2 and port 50005: No route to host
[1780696198] HOST ALERT: MTPC;DOWN;HARD;4;connect to address 172.28.54.2 and port 50005: No route to host

Relevant parent host log:

[1780695999] HOST ALERT: FW01;DOWN;SOFT;1;PING CRITICAL - Packet loss = 100%
[1780696062] HOST ALERT: FW01;UP;SOFT;1;PING OK - Packet loss = 0%, RTA = 10.60 ms

Resulting hoststatus from status.dat:

current_state=1
last_hard_state=1
current_problem_id=0
last_problem_id=647643
current_attempt=4
max_attempts=4
state_type=1
last_state_change=1780696195
last_hard_state_change=1780696195
last_time_up=1780695999
last_time_down=1780697698
last_time_unreachable=1780696062
problem_has_been_acknowledged=0
scheduled_downtime_depth=0
notifications_enabled=1
current_notification_number=2

retention.dat shows the same current_problem_id=0 and last_problem_id=647643.

No passive host check result was found:

grep 'PROCESS_HOST_CHECK_RESULT;MTPC' /usr/local/nagios/var/nagios.log

returned no results.

Suspected trigger:
The child host first entered UNREACHABLE/SOFT because the parent was temporarily DOWN/SOFT. The parent recovered before the child reached hard state. The child was then reclassified from UNREACHABLE/SOFT to DOWN/SOFT and eventually DOWN/HARD, but current_problem_id remained 0.

Impact:
Service-Now automations that rely on HOSTPROBLEMID/current_problem_id to open or correlate incidents miss valid HARD host problems because current_problem_id remains 0 even though the host is DOWN/HARD and notifications are sent.

-----
Parent:
DOWN/SOFT;1 -> UP/SOFT;1

Child:
UNREACHABLE/SOFT;1 -> DOWN/SOFT;2 -> DOWN/SOFT;3 -> DOWN/HARD;4

Result:
current_state=1
state_type=1
current_problem_id=0
last_problem_id=<non-zero>
